lib/gyp/generator/ninja/index.js

```diff
@@ -140,6 +143,8 @@ Ninja.prototype.expand = function expand(p, productDir) {
     p = p.replace(/\$!INTERMEDIATE_DIR/g, intDir);
   }
 
+  // First one is in order to outsmart silly GYP file writers
+  p = p.replace(/(\.\.[\\\/])+\$\|CONFIGURATION_NAME/g, productDir);
```

Comment:
So the `GYP` is:
```
{
              # build final .dat -> .obj
              'action_name': 'genccode',
              'inputs': [ '<(SHARED_INTERMEDIATE_DIR)/icutmp/icudt<(icu_ver_major)<(icu_endianness).dat' ],
              'outputs': [ '<(SHARED_INTERMEDIATE_DIR)/icudt<(icu_ver_major)<(icu_endianness)_dat.obj' ],
              'action': [ '../../<(CONFIGURATION_NAME)/genccode',
                         '-o',
                         '-d', '<(SHARED_INTERMEDIATE_DIR)/',
                         '-n', 'icudata',
                         '-e', 'icusmdt<(icu_ver_major)',
                         '<@(_inputs)' ],
}
```
The result (pre patch) is:
```
rule genccode
  command = cmd.exe$ /s$ /c$ "cd$ ..\..\tools\icu$ &$ ..\..\Release\genccode$ $
    -o$ -d$ ..\..\out\Release\gen\$ -n$ icudata$ -e$ icusmdt58$ $
    ..\..\out\Release\gen\icutmp\icudt58l.dat"
```

They are trying to reach the output dir, but the path voodoo fails, `..\..\Release\genccode` is not there, it's at `..\..\out\Release\genccode`

so after patch:
```
rule genccode
  command = cmd.exe$ /s$ /c$ "cd$ ..\..\tools\icu$ &$ $
    ..\..\out\Release\genccode$ -o$ -d$ ..\..\out\Release\gen\$ -n$ icudata$ $
    -e$ icusmdt58$ ..\..\out\Release\gen\icutmp\icudt58l.dat"
```

I make a PR to fix the root nodejs/node#11217